nucypher/blockchain/eth/actors.py

```diff
@@ -1358,3 +1374,81 @@ def total_stake(self) -> NU:
         stake = sum(self.staking_agent.owned_tokens(staker_address=account) for account in self.wallet.accounts)
         nu_stake = NU.from_nunits(stake)
         return nu_stake
+
+
+class Bidder(NucypherTokenActor):
+    """WorkLock participant"""
+
+    class BidingIsClosed(RuntimeError):
+        pass
+
+    def __init__(self, checksum_address: str, *args, **kwargs):
+        super().__init__(checksum_address=checksum_address, *args, **kwargs)
+        self.worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=self.registry)
+        self.staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=self.registry)
+
+    def ensure_bidding_is_open(self):
+        now = maya.now().epoch
+        start = self.worklock_agent.start_date
+        end = self.worklock_agent.end_date
+        if now < start:
+            raise self.BiddingIsClosed(f'Bidding does not open until {maya.MayaDT(start).slang_date()}')
+        if now > end:
+            raise self.BiddingIsClosed(f'Bidding closed at {maya.MayaDT(end).slang_date()}')
+
+    #
+    # Transactions
+    #
+
+    def place_bid(self, value: int) -> dict:
+        # wei_bid = Web3.toWei(value, 'wei')  # TODO: Use this?
+        self.ensure_bidding_is_open()
+        receipt = self.worklock_agent.bid(checksum_address=self.checksum_address, value=value)
+        return receipt
+
+    def claim(self) -> dict:
+        receipt = self.worklock_agent.claim(checksum_address=self.checksum_address)
+        return receipt
+
+    def cancel_bid(self) -> dict:
+        receipt = self.worklock_agent.cancel_bid(checksum_address=self.checksum_address)
+        return receipt
+
+    def refund_deposit(self):
+        """Refund ethers for completed work"""
+        receipt = self.worklock_agent.refund(checksum_address=self.checksum_address)
+        return receipt
```

Comment:
Checks that can be added: there is still not refunded eth, `self.available_refund > 0`